internal/controller/ols_app_server_reconciliator.go

```diff
@@ -240,3 +243,38 @@ func (r *OLSConfigReconciler) reconcileService(ctx context.Context, cr *olsv1alp
 	r.logger.Info("OLS service reconciled", "service", service.Name)
 	return nil
 }
+
+func (r *OLSConfigReconciler) reconcileLLMSecrets(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
+	providerCredentials := ""
+	var providerApiToken string
+	for _, provider := range cr.Spec.LLMConfig.Providers {
+		foundSecret := &corev1.Secret{}
+		err := r.Client.Get(ctx, client.ObjectKey{Name: provider.CredentialsSecretRef.Name, Namespace: r.Options.Namespace}, foundSecret)
+		if err != nil {
+			return fmt.Errorf("failed to fetch provider secret: %w", err)
+		}
+		providerApiToken, err = getSecretContent(r.Client, provider.CredentialsSecretRef.Name, r.Options.Namespace, LLMApiTokenFileName)
+		if err != nil {
+			return fmt.Errorf("failed fetching token for provider: %s. error: %w", provider.Name, err)
+		}
+		providerCredentials += providerApiToken
+		if err = controllerutil.SetControllerReference(cr, foundSecret, r.Scheme); err != nil {
+			return fmt.Errorf("failed to set controller reference to secret: %s. error: %w", foundSecret.Name, err)
+		}
+		err = r.Update(ctx, foundSecret)
```

Comment:
@vishnuchalla Could you please help understand , the need for this Update here ? Not seeing any updates to Secret.

Reply:
This is used to update the secrets with controller reference, so that they can be tracked in our reconciliation process.